credential/signature.go

```diff
@@ -0,0 +1,110 @@
+package credential
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/pkg/errors"
+)
+
+// VerifyCredentialSignature verifies the signature of a credential of any type
+// TODO(gabe) support other types of credentials https://github.com/TBD54566975/ssi-sdk/issues/352
+func VerifyCredentialSignature(ctx context.Context, genericCred any, resolver did.Resolver) (bool, error) {
+	if genericCred == nil {
+		return false, errors.New("credential cannot be empty")
+	}
+	if resolver == nil {
+		return false, errors.New("resolver cannot be empty")
+	}
+	switch genericCred.(type) {
+	case *VerifiableCredential:
+		return VerifyCredentialSignature(ctx, *genericCred.(*VerifiableCredential), resolver)
+	case VerifiableCredential:
+		cred := genericCred.(VerifiableCredential)
```

Comment:
no ok checking here?
cred, ok := genericCred.(VerifiableCredential)
I guess we know the type so not needed

Reply:
yeah since the switch was hit we know the type, I can make this nicer by extracting the type above

Reply:
I'm not sure how to make it nicer without some crazy generic stuff